src/pages/user/SignupForm.jsx

```diff
@@ -0,0 +1,83 @@
+import { useState } from "react";
+import { useNavigate } from "react-router";
+
+export default function SignupForm({ setAuthUser }) {
+  const navigate = useNavigate();
+
+  const [user, setUser] = useState({
+    username: "",
+    password: "",
+  });
+
+  const handleInputChange = (event) => {
+    console.log(event.target.name, event.target.value);
+
+    const inputName = event.target.name;
+    const inputValue = event.target.value;
+
+    setUser({ ...user, [inputName]: inputValue });
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+
+    const userToCreate = {
+      ...user,
+    };
+
+    const fetchOptions = {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(userToCreate),
+    };
+
+    fetch(`${process.env.REACT_APP_API_URL}/signup`, fetchOptions)
+      .then((res) => {
+        if (res.status === 400) {
+          throw Error("Missing information!");
+        } else if (res.status !== 201) {
+          throw Error(res);
```

Comment:
Passing through `res` isn't going to be too helpful. Use a message like `[401 ERROR] Unauthorized` or `[500 ERROR] Internal Server Error` instead.